### PR TITLE
Add style blocks to body rather than head v2

### PR DESCRIPTION
**Pull request by Renkas**
This is modification of existing pull request: #359 

I removed the option to move style element to body and made it default/only way.
Added unit test and modified some existing ones.

### CODE REVIEW

**Comment by oliverklee** [comment]
@Renkas Any news on this PR? (Please don't feel discouraged by the review comments. I aim to get your changes into the code base in their best form.)

File: Classes/Emogrifier.php [hunk 1]
```diff
@@ -1037,8 +1037,10 @@ protected function addStyleElementToDocument(\DOMDocument $document, $css)
         $styleAttribute->value = 'text/css';
         $styleElement->appendChild($styleAttribute);
 
-        $head = $this->getOrCreateHeadElement($document);
-        $head->appendChild($styleElement);
+        $body = $this->getBodyElement($document);
+        if ($body) {
+            $body->appendChild($styleElement);
+        }
```
[the remark this answers is not in the document]
Comment (Renkas) [reply Classes/Emogrifier.php]: Only way there would not be a body tag is if there is no content in document. In all other cases DOMDocument will add the body tag automatically. But document without content is useless anyway.

File: Classes/Emogrifier.php [hunk 2]
```diff
@@ -1037,8 +1037,10 @@ protected function addStyleElementToDocument(\DOMDocument $document, $css)
         $styleAttribute->value = 'text/css';
         $styleElement->appendChild($styleAttribute);
 
-        $head = $this->getOrCreateHeadElement($document);
```
[the remark this answers is not in the document]
Comment (Renkas) [reply Classes/Emogrifier.php]: It is removed

File: Classes/Emogrifier.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (oliverklee) [reply Classes/Emogrifier.php]: Are you sure?

As far as I know, the HEAD and BODY elements are optional, i.e., the child tags (TITLE etc. for HEAD, H1 etc. for BODY) can be direct children of the HTML element. Does DOMDocument then automatically sort the existing tags correctly in either HEAD or BODY, or will e.g., TITLE also go into BODY?

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 ## x.y.z (unreleased)
 
 ### Added
+- Added setting for putting the media query style block to <body> vs <head>
```
Comment (oliverklee) [inline CHANGELOG.md]: I propose we simplify/shorten this to: "Automatically the STYLE block into the BODY". What do you think=

**Comment by oliverklee** [comment]
We've moved the repository to https://github.com/MyIntervals/emogrifier . If you have an "upstream" branch in your fork, please make sure to update the URL:

`git remote set-url upstream git@github.com:MyIntervals/emogrifier.git`

**Comment by Renkas** [comment]
So what is the status?

**Comment by oliverklee** [comment]
I seems that you have re-pushed. Unfortunately. GitHub does not send any notifications of this. :-(

Thanks for the reminder! :cookie: I'll have another look at the PR.

**Comment by oliverklee** [comment]
I've just created #410 for making sure we always have a BODY element.

**Comment by oliverklee** [comment]
@Renkas Now that #410 is merged (which makes sure we always have a BODY element), could you please rebase and make use of the BODY element always being there? Thanks!

**Comment by oliverklee** [comment]
Merged. Thanks for contributing, @Renkas and @Gabe-SpotMe! :heart:
